pkg/pillar/cmd/zedagent/handleconfig.go

```diff
@@ -528,12 +619,12 @@ func requestConfigByURL(getconfigCtx *getconfigContext, url string,
 	// except http.StatusForbidden(which returns error
 	// irrespective of bailOnHTTPErr)
 	getconfigCtx.configGetStatus = types.ConfigGetFail
-	b, cr, err := generateConfigRequest(getconfigCtx)
+	b, err := generateConfigRequest(getconfigCtx, isCompoundConfig)
 	if err != nil {
 		log.Fatal(err)
 	}
 	buf := bytes.NewBuffer(b)
-	size := int64(proto.Size(cr))
+	size := int64(buf.Len())
```

Comment:
How different is the proto.Size and buf.Len? Didn't see the motivation for the change and I think we use proto.Size elsewhere.

Reply:
the motivation is to eliminate the return value, which is used only for calculating the size, which should be equal to already parsed buffer, which we return from the same function. (if I understand how the protobuf works)

Reply:
You're right - didn't see that context.
But can you check if we use proto.Size equally unecessarily elsewhere in zedagent?